pkg/kb-summarizer/summarizer/summarizer.go

```diff
@@ -360,10 +377,58 @@ func (s *Summarizer) save() error {
 	jsonWriter := io.Writer(jsonFile)
 	encoder := json.NewEncoder(jsonWriter)
 	encoder.SetIndent("", " ")
+
```

Comment:
Please add some unit tests for the new logic. I understand we are very thin on existing tests across the repo, so as we add more code we should try to expand on that.

To make life easier, maybe (part of) the new logic could be extracted on its own `func`.

Reply:
done